crates/bevy_render/src/render_resource/bind_group.rs

```diff
@@ -353,6 +353,8 @@ pub enum AsBindGroupError {
     /// The bind group could not be generated. Try again next frame.
     #[error("The bind group could not be generated")]
     RetryNextUpdate,
+    #[error("The bindgroup data is invalid: {0}")]
+    InvalidData(String),
```

Comment:
I think it would be better to avoid too broad terms, since `InvalidData` can be anything. In this case, the new error is added specifically for when missing or invalid sampler, so it could be something in the likes of `InvalidSampler`